.github/workflows/tf-test-deployment.yml

```diff
@@ -0,0 +1,436 @@
+          echo "cid_cfn_version=$CFN_VERSION" >> $GITHUB_OUTPUT
+          
+          # Get Data Export version
+          EXPORT_VERSION=$(curl -s https://raw.githubusercontent.com/aws-solutions-library-samples/cloud-intelligence-dashboards-data-collection/main/data-exports/deploy/data-exports-aggregation.yaml | grep Description | grep -o '[0-9]\+\.[0-9]\+\.[0-9]\+' | head -1)
+          echo "data_export_version=$EXPORT_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Create terraform.tfvars
+        run: |
+          cat > ./terraform/terraform.tfvars << EOF
+          global_values = {
+            destination_account_id = "${{ secrets.MANAGEMENT_ACCOUNT_ID }}"
+            source_account_ids     = "${{ secrets.MANAGEMENT_ACCOUNT_ID }}"
+            aws_region             = "${{ secrets.AWS_REGION }}"
+            quicksight_user        = "${{ secrets.QUICKSIGHT_USER }}"
+            cid_cfn_version        = "${{ steps.versions.outputs.cid_cfn_version }}"
+            data_export_version    = "${{ steps.versions.outputs.data_export_version }}"
+            environment            = "dev"
+          }
+          EOF
+
+      - name: Create provider override for single-account deployment
+        run: |
+          cat > ./terraform/local_override.tf << EOF
+          provider "aws" {
+            alias = "destination_account"
+            # This will use the same credentials as the default provider
+          }
+          EOF
+
+      - name: Override backend.tf with S3 backend
+        run: |
+          cat > ./terraform/backend.tf << EOF
+          terraform {
+            backend "s3" {
+              bucket = "${{ secrets.BACKEND_S3_BUCKET }}"
+              key    = "terraform/cid-test/terraform.tfstate"
+              region = "${{ secrets.AWS_REGION }}"
+            }
+          }
+          EOF
+
+      # First setup and validate the original code
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.MANAGEMENT_ROLE_ARN }}
+          aws-region: eu-west-2
+          role-duration-seconds: 3600
+          role-skip-session-tagging: true
+
+      - name: Initialize Terraform for validation
+        working-directory: ./terraform
+        run: terraform init
+
+      - name: Setup TFLint
+        uses: terraform-linters/setup-tflint@v3
+
+      - name: Run TFLint
+        working-directory: ./terraform
+        run: tflint --format=compact
+
+      - name: Terraform Format Check
+        working-directory: ./terraform
+        run: terraform fmt -check -recursive
+
+      - name: Terraform Validate
+        working-directory: ./terraform
+        run: terraform validate
+
+      # Now modify the files for testing
+      - name: Modify Terraform files for single-account deployment
+        run: |
+          # Create a temporary directory for modified Terraform files
```

Comment:
quite a big script. worth adding as a separate file? also for local usage can be useful?